### PR TITLE
Neaten endpoint tests draft

### PR DESCRIPTION
**Pull request by jeffplays2005**
Just to check and see if everything will still work.

Created a `setupTests.ts` file to include most of the generic code that's repeated or unecessary in most of the files for the future.

Also added the setup file to `jest.config.server.ts`

Removed redundant setup code from `routes.test.ts`

### CODE REVIEW

**Comment by jeffplays2005** [comment]
just a comment, this is far from being done, just chucking it here to see if anything goes wrong from GitHub tests(tested fine on my laptop).

**Comment by jeffplays2005** [comment]
Still a bit far from being done, just need to do some editing to individual routes and also figure out if the setup files can be done any better. 

All the routes are just a 1 to 1 copy from the original `routes.test.ts` without any neatening.

File: package.json [hunk 1]
```diff
@@ -54,9 +54,10 @@
     "dev-client": "yarn workspace client dev",
     "dev-server": "yarn workspace server start",
     "storybook": "yarn workspace client storybook",
-    "test-server": "yarn firebase-test \"node --experimental-vm-modules --dns-result-order=ipv4first ./node_modules/jest/bin/jest.js --setupFiles dotenv/config --passWithNoTests --detectOpenHandles --forceExit --testPathPattern=server/src\"",
+    "test-endpoints": "yarn firebase-test \"node --experimental-vm-modules --dns-result-order=ipv4first ./node_modules/jest/bin/jest.js --setupFiles dotenv/config --passWithNoTests --detectOpenHandles --forceExit --testPathPattern=server/src/middleware --config=server/jest.config.endpoints.ts\"",
+    "test-server": "yarn firebase-test \"node --experimental-vm-modules --dns-result-order=ipv4first ./node_modules/jest/bin/jest.js --setupFiles dotenv/config --passWithNoTests --detectOpenHandles --forceExit --testPathPattern=server/src --config=server/jest.config.server.ts\"",
```
Comment (bcho892) [inline package.json:58]: nit: if this is exact same you can make a separate script like `jest-with-setup` and then do `yarn jest-with-setup --config=server/jest.config.server.ts\`

File: server/jest.config.server.ts [hunk 2]
```diff
@@ -6,5 +6,6 @@ module.exports = {
   modulePaths: ["<rootDir>"],
   moduleDirectories: ["node_modules", "src"],
   setupFiles: ["dotenv/config"],
-  testMatch: ["**/*.test.ts"]
+  testMatch: ["**/*.test.ts"],
+  testPathIgnorePatterns: ["<rootDir>/src/middleware"]
```
Comment (bcho892) [inline server/jest.config.server.ts:10]: smart!

**Review by bcho892 — APPROVED** [review]

**Comment by jeffplays2005** [comment]
Testing endpoints:
```
yarn test-endpoints```
Testing other server scripts
```
yarn test-server```

**Pull request merged**